### PR TITLE
feat: trigger AE fn

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**

### CODE REVIEW

File: reya-rust-sdk/src/http_provider.rs [hunk 1]
```diff
@@ -392,6 +392,57 @@ impl HttpProvider {
         eyre::Ok(receipt)
     }
 
+    pub async fn trigger_auto_excahnge(
+        &self,
+        signer: PrivateKeySigner,
+        params: data_types::TriggerAutoExchangeParams,
+    ) -> eyre::Result<TransactionReceipt> // return the transaction receipt
```
Comment (moboware) [inline reya-rust-sdk/src/http_provider.rs:399]: In develop branch this is changed to return the tx_hash, receipts can be retrieved with a separate call.

File: reya-rust-sdk/src/http_provider.rs [hunk 2]
```diff
@@ -392,6 +392,57 @@ impl HttpProvider {
...
+        };
+
+        let builder = proxy.triggerAutoExchange(inputs);
+        // todo: change accordingly
+        let new_gas_limit = (builder.estimate_gas().await.unwrap() * 12u128) / 10u128;
+        let b2 = builder.gas(new_gas_limit);
+
+        let transaction_result = b2.send().await?;
+
+        let receipt = transaction_result.get_receipt().await?;
```
Comment (moboware) [inline reya-rust-sdk/src/http_provider.rs:435]: return the tx_hash only

**Review by moboware — CHANGES_REQUESTED** [review]
Is there a reason why you want/need to merge to main? There is now a develop branch that is ahead of main

File: reya-rust-sdk/src/http_provider.rs (hunk 1 above)
Comment (ioanaroceanu2000) [reply reya-rust-sdk/src/http_provider.rs]: resolved in [098a64f](https://github.com/Reya-Labs/reya-rust-sdks-monorepo/pull/23/commits/098a64f58f4b9760e1ead716c1a7e27e95f83f5f)

**Review by moboware — APPROVED** [review]

**Pull request merged**